pkg/k8s/init.go

```diff
@@ -153,6 +163,8 @@ func WaitForNodeInformation(ctx context.Context, k8sGetter k8sGetter) error {
 			logfields.K8sNodeIP:        k8sNodeIP,
 		}).Info("Received own node information from API server")
 
+		useNodeIPAMAllocPools(n)
```

Comment:
Would it be possible to write the reconciler in a way that we don't have to do this here? As in, do we strictly depend on having this information available early on, and if so, why?

Reply:
> do we strictly depend on having this information available early on

I don't think the allocation pools are needed this early. Let me test and I'll update the PR or provide additional feedback.

Reply:
Commit 0329660 removed this change.